lxd/container_lxc.go

```diff
@@ -365,15 +365,47 @@ func (c *containerLXC) initLXC() error {
 		}
 	}
 
-	// Setup limits
-	limitMemory, ok := c.expandedConfig["limits.memory"]
-	if ok && limitMemory != "" {
+	// Memory limits
+	limitMemory := c.expandedConfig["limits.memory"]
+	if limitMemory != "" {
 		err = lxcSetConfigItem(cc, "lxc.cgroup.memory.limit_in_bytes", limitMemory)
 		if err != nil {
 			return err
 		}
 	}
 
+	// CPU limits
+	cpuPriority := c.expandedConfig["limits.cpu.priority"]
+	cpuAllowance := c.expandedConfig["limits.cpu.allowance"]
+
+	if cpuPriority != "" || cpuAllowance != "" {
+		cpuShares, cpuCfsQuota, cpuCfsPeriod, err := deviceParseCPU(cpuAllowance, cpuPriority)
+		if err != nil {
+			return err
+		}
+
+		if cpuShares != "1024" {
+			err = lxcSetConfigItem(cc, "lxc.cgroup.cpu.shares", cpuShares)
```

Comment:
Those are the startup values so it doesn't matter there.

The live updating happens in Update() and unconditionally updates the 3 files so that live updates work as expected.